tests/neg/i2672.scala

```diff
@@ -0,0 +1,16 @@
+class Foo[T]
+
+trait Prepend {
+  type Out
+}
+
+object Test {
+  def foo()(implicit ev: Prepend): Foo[ev.Out] = ???
+
+  def test: Unit = {
+    foo(): Foo[Any] // error: found: Prepend => Foo[_] required: Foo[Any]
```

Comment:
This isn't a great error message by the way, since it might be displayed instead of a more informative message based on`@implicitNotFound`.